### ISSUE TITLE
undefined reference to `SSL_get_peer_certificate` (openssl-0.10.64)

### ISSUE THREAD

**Issue by WillDera** [sfackler/rust-openssl#2246]
I'm building a RESTAPI service with axum and its tls_openssl (for ssl support) previously this worked fine, but after updating rust and cargo via `rustup update`. I suddenly get the error in the title. Below are a few screenshots.

Here it shows the `undefined reference` error originating from the openssl crate. 
![Screenshot from 2024-05-28 06-40-26](https://github.com/sfackler/rust-openssl/assets/30760648/9427873b-fce8-4095-b93a-71a91d3a0138)

This is what my axum code looks like
![Screenshot from 2024-05-28 06-41-13](https://github.com/sfackler/rust-openssl/assets/30760648/833da81a-7dab-4420-9a84-898b7787d4c8)
![Screenshot from 2024-05-28 06-41-59](https://github.com/sfackler/rust-openssl/assets/30760648/0b8a7502-9fe9-430e-ae82-e52fb6bd6d09)

I'm using the following version: 
axum version 0.7.3 
axum-server version 0.6.0

On my machine I have openssl version 3.3.0.

I would really appreciate any help, I've been on this for a whole day and can't figure out how to fix it.

**Comment by WillDera** [comment]
Hmm, I see. I'd look into that now

**Comment by sfackler** [comment]
All openssl 1.x versions have been at end of life for months to years at this point, so I would hope that nothing is depending on them.

**Comment by WillDera** [comment]
Yeah, I saw that while researching on version 1.x last night. Well, pacman is the last resort here, but I would try messing around with services on my machine for a few days to see if anything would scream before I ultimately try to uninstall them. 

Thank you very much for your assistance.

**Comment by better629** [comment]
@WillDera 
```
  = note: /text-generation-inference/target/release/deps/libopenssl-82592f42675f9707.rlib(openssl-82592f42675f9707.openssl.312e7bf4ad1b4be1-cgu.00.rcgu.o): In function `openssl::ssl::SslRef::peer_certificate':
          openssl.312e7bf4ad1b4be1-cgu.00:(.text._ZN7openssl3ssl6SslRef16peer_certificate17h3fc79bba5c9bfba5E+0x3): undefined reference to `SSL_get_peer_certificate'
          collect2: error: ld returned 1 exit status
```

there only one `openssl` under `/usr/include`. Do you have any other ideas?
